assets/js/components/notifications/FirstPartyModeSetupBanner.test.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * FirstPartyModeSetupBanner component tests.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetchMock from 'fetch-mock';
+
+/**
+ * Internal dependencies
+ */
+import FirstPartyModeSetupBanner from './FirstPartyModeSetupBanner';
+import {
+	createTestRegistry,
+	fireEvent,
+	provideModules,
+	provideSiteInfo,
+	provideUserInfo,
+	render,
+	waitFor,
+} from '../../../../tests/js/test-utils';
+import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../googlesitekit/constants';
+import { DEFAULT_NOTIFICATIONS } from '../../googlesitekit/notifications/register-defaults';
+import { CORE_NOTIFICATIONS } from '../../googlesitekit/notifications/datastore/constants';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { withNotificationComponentProps } from '../../googlesitekit/notifications/util/component-props';
+import { enabledFeatures } from '../../features';
+
+const FPM_SETUP_BANNER_NOTIFICATION = 'first-party-mode-setup-cta-banner';
+
+describe( 'FirstPartyModeSetupBanner', () => {
+	let registry;
+
+	const notification = DEFAULT_NOTIFICATIONS[ FPM_SETUP_BANNER_NOTIFICATION ];
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		enabledFeatures.add( 'firstPartyMode' );
+
+		provideSiteInfo( registry );
+		provideUserInfo( registry );
+		provideModules( registry, [
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+			{
+				slug: 'ads',
+				active: true,
+				connected: true,
+			},
+		] );
+
+		registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+			isEnabled: false,
+			isFPMHealthy: true,
+			isScriptAccessEnabled: true,
+		} );
+
+		registry
+			.dispatch( CORE_NOTIFICATIONS )
+			.registerNotification(
+				FPM_SETUP_BANNER_NOTIFICATION,
+				notification
+			);
+
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+	} );
+
+	describe( 'checkRequirements', () => {
+		it( 'is active', async () => {
```

Comment:
Let's be slightly more explicit e.g.

```suggestion
		it( 'is active when all required conditions are met', async () => {
```